src/test/kotlin/org/pac4j/vertx/handler/impl/Functions.kt

```diff
@@ -65,7 +64,7 @@ fun logoutHandler(vertx: io.vertx.core.Vertx): Handler<RoutingContext> {
 }
 
 
-fun getProfileManager(rc: RoutingContext): VertxProfileManager<TestOAuth1Profile> {
```

Comment:
I like Groovy too, and wish Spock was a good fit for vert.x integration tests! I'm still experimenting with Kotlin a little and not sure if my test code is most efficient, but I do think the use of Kotlin has made it more legible.

I"m not so sure about Kotlin in the main codebase. I think most of the pain in the main vertx-pac4j codebase is around successfully integrating the blocking and non-blocking parts and I don't think language makes much of a difference to that complexity. If I see somewhere where I think Kotlin would genuinely help clarity and legibility then I'll consider it (and discuss with you with actual code comparisons). 

I was also considering using Lombok to reduce boilerplate around constructors/getters/setters in the Java code. However, it doesn't fit with the vert.x fluent API conventions on, for example, the options classes, so I think that's a non-starter, which is a shame.